src/scheduler/pct.rs

```diff
@@ -100,6 +97,13 @@ impl Scheduler for PctScheduler {
         // > enables/creates a second thread.
         // TODO is this really correct? need to think about it more
         if runnable.len() > 1 {
+            // If any new tasks were created, assign them priorities at random
+            let known_tasks = self.priority_queue.len();
+            let max_tasks = usize::from(*runnable.iter().max().unwrap());
+            for tid in known_tasks..max_tasks {
+                let index = self.rng.gen_range(1, self.priority_queue.len() + 1);
```

Comment:
Why is this offset by one? (a) the new task should be allowed to be top priority (b) `insert` should panic if given an index of `len + 1`

Reply:
`gen_range(l, h)` returns a value in `[l, h)`  so the max the above will return is `len`.
As for the lower bound, yeah, I agree we should allow it to be 0.  Fixing.